response_operations_ui/templates/collection_exercise/ce-ci-section.html

```diff
@@ -28,26 +28,25 @@
         %}
     {% do collectionInstrumentTableData | setAttribute("ths", tblHeaders) %}
   {% endif %}
-  
     {% for ci_type in ci_table_context["ci_details"] %}
-      {% do collectionInstrumentTableDataRows.append(
-          {
-              "tds": [
-                {
-                  "value": ci_type.title
-                },
-                {
-                  "value": ci_type.count,
-                  "tdClasses": "ons-u-ta-centre"
-                }, 
-                {
-                  "value": '<a href="' + ci_type.url + '" id="view-add-upload-ci-' + ci_type.type + '">' + ci_type.link_text + '</a>',
-                  "tdClasses": "ons-u-ta-right"
-                }
-              ] 
-          }
-      )
-    %}
+        {% do collectionInstrumentTableDataRows.append(
+            {
+                "tds": [
+                  {
+                    "value": ci_type.title
+                  },
+                  {
+                    "value": ci_type.count,
+                    "tdClasses": "ons-u-ta-centre"
+                  }, 
+                  {
+                    "value": '<a href="' + url_for('collection_exercise_bp.get_seft_collection_instrument', period=ce.exerciseRef, short_name=survey.shortName.replace(' ', '')) + '" id="seft-upload-link">' + ci_type.link_text + '</a>' if ci_type.type == "seft" else '<a href="' + ci_type.url + '" id="view-add-upload-ci-' + ci_type.type + '">' + ci_type.link_text + '</a>',
```

Comment:
This could and probably should be done in the backend. The href is only a url, id and link text, you just need to do the logic in the backend and pass them through (url and link_text is already passed) In fact you can get rid of type from the dict if you do, or just use the same id for seft (view-add-upload-ci-)

Reply:
Have moved this logic to the backend. Used the same id for seft and eq however acceptance tests cannot find the id when run as a seft so working through that now.